### DISCUSSION TITLE
Enable "Liking" items within search results

### DISCUSSION THREAD

**Discussion by jftrim** [microsoft-search/pnp-modern-search#1617] [Q&A] [answered]
I am searching across a Sharepoint list which has user ratings enabled.  I am displaying the number of "Likes" in the search results, but it would be great to be able to upvote items directly from the results list.

Is there a way to do this using Custom code in the results Styling section?

**Comment by wobba** [comment, marked as answer]
You would need to create a custom web component and use that in your templates to achieve this. https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/

**Reply by jftrim** [reply]
Thanks you  - this is a bit beyond my skills, but good to know it's feasible

**Reply by huetvictoria** [reply]
Hi @jftrim - I am also trying to display number of 'Likes' in my Search Results. I would really appreciate any guidance you can provide on how you did this (i.e., the custom template you used for the display). Thank you so much!

**Comment by simesy** [comment]
It would be handy if you could either:

1. Return a yes/no property that indicated if the current user liked an item, or
2. Add a refiner to restrict results to what the user has liked.

I can't find how to do this with the Graph (let alone PnP Search). I found some old issues from SP 2013 but that's the closest I've got.

**Comment by simesy** [comment]
(I realise I'm slightly hijacking this thread but hopefully I can help others in their googling journeys.)

For example I want to work out if I have to do a call back on every item (which of course would be possible see screenshot) just to know if it is liked by the current user for filtering or display. If you have 50 items are you going to callback to the server fifty times? Ideally the Graph could be populating this server-side.

<img width="1296" alt="Screen Shot 2022-04-22 at 5 41 47 pm" src="https://user-images.githubusercontent.com/188856/164639678-0c13c862-cfbb-49b0-9a73-834a69bbeff8.png">

**Reply by wobba** [reply]
Graph support batch queries - optimizing the call pattern.

**Reply by simesy** [reply]
Definitely feels like a custom web part is the way to go (notwithstanding ways to extend PnP Search).
